src/device/pf_pdport.c

```diff
@@ -228,69 +283,92 @@ static void pf_pdport_remove_all_diag (pnet_t * net)
 {
-   if (pf_pdport_load (net) != 0)
+   int port;
+   pf_port_iterator_t port_iterator;
+
+   pf_port_init_iterator_over_ports (net, &port_iterator);
+   port = pf_port_get_next (&port_iterator);
+   while (port != 0)
    {
-      /* Create file if missing */
-      (void)pf_pdport_save (net);
+      if (pf_pdport_load (net, port) != 0)
+      {
+         /* Create file if missing */
+         (void)pf_pdport_save (net, port);
+      }
+
+      port = pf_port_get_next (&port_iterator);
    }
    return 0;
 }
 
 int pf_pdport_reset_all (pnet_t * net)
 {
+   int port;
+   pf_port_iterator_t port_iterator;
+   pf_port_t * p_port_data = NULL;
    const char * p_file_directory = NULL;
-   pf_port_t * p_port_data = pf_port_get_state (net, PNET_PORT_1);
 
    (void)pf_cmina_get_file_directory (net, &p_file_directory);
-   pf_file_clear (p_file_directory, PF_FILENAME_PDPORT);
 
-   memset (&p_port_data->pdport, 0, sizeof (p_port_data->pdport));
-   pf_pdport_remove_all_diag (net);
+   pf_port_init_iterator_over_ports (net, &port_iterator);
+   port = pf_port_get_next (&port_iterator);
+   while (port != 0)
+   {
+      p_port_data = pf_port_get_state (net, port);
+      memset (&p_port_data->pdport, 0, sizeof (p_port_data->pdport));
+      pf_pdport_remove_all_diag (net, port);
+      pf_file_clear (p_file_directory, pf_pdport_get_filename (port));
 
+      port = pf_port_get_next (&port_iterator);
+   }
    return 0;
 }
 
-void pf_pdport_ar_connected (pnet_t * net, const pf_ar_t * p_ar)
+void pf_pdport_remove_data_files (const char * file_directory)
 {
-   /* Todo check which dap slots are active in ar and map to ports */
-
-   pf_port_t * p_port_data = pf_port_get_state (net, PNET_PORT_1);
-   if (p_port_data->pdport.adjust.active == true)
+   int port;
+   for (port = PNET_PORT_1; port <= PNET_MAX_PORT; port++)
    {
-      if (
-         p_port_data->pdport.adjust.peer_to_peer_boundary.peer_to_peer_boundary
-            .do_not_send_LLDP_frames == 1)
-      {
-         pf_lldp_send_disable (net, PNET_PORT_1);
-      }
-      else
-      {
-         pf_lldp_send_enable (net, PNET_PORT_1);
-      }
+      pf_file_clear (file_directory, pf_pdport_get_filename (port));
    }
 }
 
+void pf_pdport_ar_connected (pnet_t * net, const pf_ar_t * p_ar)
+{
+   (void)p_ar;
+   pf_pdport_lldp_restart (net);
+}
+
 void pf_pdport_lldp_restart (pnet_t * net)
 {
-   /* Todo check which dap slots are active in ar and map to ports */
+   int port;
+   pf_port_iterator_t port_iterator;
+   pf_port_t * p_port_data = NULL;
+
+   pf_port_init_iterator_over_ports (net, &port_iterator);
+   port = pf_port_get_next (&port_iterator);
 
-   pf_port_t * p_port_data = pf_port_get_state (net, PNET_PORT_1);
-   if (p_port_data->pdport.adjust.active == true)
+   while (port != 0)
    {
-      if (
-         p_port_data->pdport.adjust.peer_to_peer_boundary.peer_to_peer_boundary
-            .do_not_send_LLDP_frames == 1)
+      if (p_port_data->pdport.adjust.active == true)
```

Comment:
We need to do `pf_port_get_state() `first?

Reply:
Yes, that was causing the unit test failure.
Done